plugins/streaming/impls/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTClientBuilder.java

```diff
@@ -242,41 +224,36 @@ public MqttClient buildAndConnect() throws MqttException {
 
     client.setCallback( callback );
 
-    logChannel.logDebug( "Subscribing to topics with a quality of service level of "
-      + variableSpace.environmentSubstitute( qos ) );
-    logChannel.logDebug( "Server URIs is set to " + variableSpace.environmentSubstitute( serverUris ) );
-    logChannel.logDebug( "Max Inflight is set to " + variableSpace.environmentSubstitute( maxInflight ) );
-    logChannel.logDebug( "Automatic Reconnect is set to " + variableSpace.environmentSubstitute( automaticReconnect ) );
+    logChannel.logDebug( "Subscribing to topics with a quality of service level of " + qos );
+    logChannel.logDebug( "Server URIs is set to " + serverUris );
+    logChannel.logDebug( "Max Inflight is set to " + maxInflight );
+    logChannel.logDebug( "Automatic Reconnect is set to " + automaticReconnect );
     logChannel.logDebug( loggableOptions().toString() );
 
     client.connect( getOptions() );
     if ( topics != null && topics.size() > 0 ) {
       client.subscribe(
-        variableSpace.environmentSubstitute( topics.toArray( new String[ topics.size() ] ) ),
-        initializedIntAray( Integer.parseInt( variableSpace.environmentSubstitute( this.qos ) ) )
+        topics.toArray( new String[ topics.size() ] ),
+        initializedIntAray( Integer.parseInt( this.qos ) )
       );
     }
     return client;
   }
 
-  private String getBroker() {
-    return variableSpace.environmentSubstitute( this.broker );
-  }
-
   private String getProtocol() {
     return isSecure ? SECURE_PROTOCOL : UNSECURE_PROTOCOL;
   }
 
   private void validateArgs() {
     // expectation that the broker will contain the server:port.
-    checkArgument( getBroker().matches( "^[^ :/]+:\\d+" ),
+    checkArgument( this.broker.matches( "^[^ :/]+:\\d+" ),
       getString( PKG, "MQTTInput.Error.ConnectionURL" ) );
     try {
-      int qosVal = Integer.parseInt( variableSpace.environmentSubstitute( this.qos ) );
+      int qosVal = Integer.parseInt( this.qos );
       checkArgument( qosVal >= 0 && qosVal <= 2 );
     } catch ( Exception e ) {
-      String errorMsg = getString( PKG, "MQTTClientBuilder.Error.QOS",
-        stepName, variableSpace.environmentSubstitute( qos ) );
+      String errorMsg = getString( PKG, "MQTTClientBuilder.Error.QOS", stepName, qos );
+      logChannel.logError( errorMsg );
```

Comment:
logError was removed in a previous commit https://github.com/pentaho/pentaho-kettle/pull/5244/files#diff-08cf9294a17b3dad77fb3375fd388bc1L280. The logging is handled in the MQTT Producer and MQTT Consumer.